crates/router/src/scheduler/producer.rs

```diff
@@ -22,9 +22,13 @@ pub async fn start_producer(
     scheduler_settings: Arc<SchedulerSettings>,
     (tx, mut rx): (mpsc::Sender<()>, mpsc::Receiver<()>),
 ) -> CustomResult<(), errors::ProcessTrackerError> {
-    use rand::Rng;
-    let timeout = rand::thread_rng().gen_range(0..=scheduler_settings.loop_interval);
-    tokio::time::sleep(std::time::Duration::from_millis(timeout)).await;
+    use rand::distributions::{Distribution, Uniform};
+
+    let mut rng = rand::thread_rng();
+    let timeout = Uniform::try_from(0..=scheduler_settings.loop_interval)
+        .into_report()
+        .change_context(errors::ProcessTrackerError::InvalidInput)?;
```

Comment:
Is this PR blocked because of this?

Reply:
@anupj Not exactly blocked by the error propagation, there's still one remaining usage of `Uniform::from()` that would have to be replaced.

https://github.com/juspay/hyperswitch/pull/1908#discussion_r1296186863

Let us know if you need any help.

Reply:
Ok, I'll fix this today/tomorrow and update PR for review. Thanks.

Reply:
@SanchithHegde I remember now why I didn't change this to try_from() because it will mean that we'll have to change the function signature for tokio_mock_sleep() to return a Result which could turn out to be a bigger change than is required for this PR as we haave to then change the fn signature where tokioo_mock_sleep() is called. Is that desirable? ?

Reply:
Can you then add a TODO here & if possible create a new issue for this that can be picked up later?